### ISSUE TITLE
Feature: Require name, email and message before starting a conversation

### ISSUE THREAD

**Issue by pranavrajs** [chatwoot/chatwoot#1131]
### Problem

In the current design, the user can start a conversation with a `Hi` or `Hey` and expects the agent to reply to the message. While it may be okay with a lot of people, it would be better for some businesses to have the complete context about the query before agent can start working on it. 

### Solution

Add an additional screen before creating the conversation which asks for the email address, name and message. This would help the user to build the properly instead of greeting messages.

### Acceptance criteria

- Add a new screen as seen in the design
- Add an option in the inbox to specify whether the business needs email, name before creating a conversation.
- Allow the user to create the message only after name & email fields are filled and valid

### Design 

<img src="https://user-images.githubusercontent.com/2246121/89738766-dca66d00-da98-11ea-8e55-bce3080add7e.png" width="500" />

**Comment by WevrLabs** [comment]
Very Nice!
How this will work in case Name and Email already been passed by the Website SDK?
I suggest in such case, to display the normal look once the user clicks the "Send Message" button, i.e: the normal look of the widget, with just minor adjustments like this:

<img width="429" alt="Screen Shot 2020-08-07 at 7 18 53 PM" src="https://user-images.githubusercontent.com/66124192/89778484-e74f1980-db0d-11ea-8871-f98c332424f6.png">

**Comment by pranavrajs** [comment]
@WevrLabs If the name, email is already available, then we will skip this screen.

**Comment by iammattmartin** [comment]
If someone chats and that information has already been passed, instead of just having the "type your message" prompt, it would still be good to preset a similar info grab as per @pranavrajs screen shot but just missing the name/email capture.

It could be as simple as "please enter your question below", or even contextual to add more of a personal feel...

"Hi [name], blah blah blah, type your question below and one of our team will help as quick as they can.

We'll respond here but also send this to your email address, [email]".

Just a better UX.

**Comment by tabrza** [comment]
In addition, if the team is offline, a similar screen could be used. In the inbox settings option the admin can select whether this screen should be show always, only if offline or never. Perhaps different header options should be possible if it were used in this case scenario.

**Comment by Giszmo** [comment]
The instant availability of a chat input is very inviting to the user but results in people just saying "Hi" to check if you reply instantly or whatever. Those people that don't really want to engage in a conversation are the ones I would like to get rid of but without scaring away those that actually have something more involved to say.

How about letting the user write his initial message and then expand the form to also ask for name and email. The Hi-guy will abandon his message but the one who explained his problem already will think twice before throwing that away. (I hate when forms do this but I also hate when they ask for my email first.)

**Comment by iammattmartin** [comment]
Agree on this. The amount of "Hi" messages that we get, reply to and then never hear anything again is crazy.

**Comment by iammattmartin** [comment]
Is there an updated release for this?

It was originally set to be in 1.8.0 but now 1.13.0 which is a good 5 releases later. 

It's nice to get some other small wins out of the way but as for as website widgets/chat goes, this really is fundamental in capturing data (and stopping silly chats with one word messages).

**Comment by pranavrajs** [comment]
@iammattmartin I understand the pain, we would definitely cover this as a part of v1.13.0 releasing on Feb 15.

**Comment by orlando** [comment]
Thanks @pranavrajs, looking forward to this feature.

**Comment by iammattmartin** [comment]
@sojan-official @pranavrajs 

I've just tested this in the latest dev repo and it appears to not show anything but also looks like it may have broken the box layout.

Seeing an extra box and formatting is off for the "type your message" input.

![Screenshot 2021-02-15 at 7 04 16 PM](https://user-images.githubusercontent.com/15971590/107984831-b968f700-6fc0-11eb-92c9-d2e67942b70c.png)

**Comment by pranavrajs** [comment]
@iammattmartin We will merge #1776 right away, it would fix this. There is a setting you need to enable for this to work.

**Comment by iammattmartin** [comment]
Ah! Will monitor that and test on my dev install.

**Comment by pranavrajs** [comment]
@iammattmartin This has been merged, please try redeploying the latest and see if it works.

**Comment by iammattmartin** [comment]
That works!

Although one thing to note, it now misses off the original intro message. So that if someone sends a message, there's nothing to explain that we'll respond within X period.

Might cause people to message multiple times without knowing what is going on.

**Comment by pranavrajs** [comment]
@iammattmartin I don't understand the intro message you mentioned, just test it out with the pre-chat form. 


<img width="449" alt="Screenshot 2021-02-16 at 1 16 40 AM" src="https://user-images.githubusercontent.com/2246121/107987760-a243fc80-6ff4-11eb-9213-1cc9a09a96ba.png">

**Comment by iammattmartin** [comment]
Hmm. I did get that but took a few minutes. Will continue testing.

**Comment by tabrza** [comment]
@pranavrajs - what about using custom input types for users depending on field types in db? for example, enabling admin to prompt user for different combos on what they might require
option a) name and phone number
option b) name, email, phone number
option c) name, phone number, message
option c) name, fb link (enabling profile pull directly into contact creation), message

**Comment by sojan-official** [comment]
@tabrza  I have logged another issue for the same.
